modin/experimental/core/execution/native/implementations/omnisci_on_native/dataframe/dataframe.py

```diff
@@ -267,37 +269,62 @@ def ref(self, col):
 
     def mask(
         self,
-        row_indices=None,
-        row_numeric_idx=None,
-        col_indices=None,
-        col_numeric_idx=None,
-    ):
+        row_labels: Optional[List[Hashable]] = None,
+        row_positions: Optional[List[int]] = None,
+        col_labels: Optional[List[Hashable]] = None,
+        col_positions: Optional[List[int]] = None,
+    ) -> "OmnisciOnNativeDataframe":
         """
-        Mask operation.
+        Mask rows and columns in the dataframe.
+
+        Allow users to perform selection and projection on the row and column labels (named notation),
+        in addition to the row and column number (positional notation).
 
         Parameters
         ----------
-        row_indices : list, optional
-            Indices of rows to select.
-        row_numeric_idx : list-like of ints, optional
-            Numeric indices of rows to select.
-        col_indices : list, optional
-            Indices of columns to select.
-        col_numeric_idx : list-like of ints, optional
-            Numeric indices of columns to select.
+        row_labels : list of hashable, optional
+            The row labels to extract.
+        row_positions : list of int, optional
+            The row positions to extract.
+        col_labels : list of hashable, optional
+            The column labels to extract.
+        col_positions : list of int, optional
+            The column positions to extract.
 
         Returns
         -------
         OmnisciOnNativeDataframe
             The new frame.
+
+        Notes
+        -----
+        If both row_labels and row_positions are provided, an exception is raised.
+        The same rule applies for col_labels and col_positions.
```

Comment:
nit: specify that a ValueError is raised. Also, add that an exception is also raised if none of col_labels, col_positions, row_labels, row_positions are provided.